onnxoptimizer/passes/fuse_cast_into_initializer.h

```diff
@@ -0,0 +1,318 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+// Before:
+//   A is in the initializer list
+//   X = Cast(A)
+//   Z = X + Y
+// After:
+//   B is in the initializer list
+//   Z = B + Y
+//
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+namespace {
+
+enum DataType {
```

Comment:
They really don't have their own enum for that?

Reply:
They do. Updates.